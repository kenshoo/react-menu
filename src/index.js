export { Menu } from "./components/menu";
