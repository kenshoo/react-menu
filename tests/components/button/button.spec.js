import React from "react";
import renderer from "react-test-renderer";
import { mount } from "enzyme";

import { Button } from "../../../src/";

describe("Button component", () => {
  test("renders children", () => {
    const cmp = renderer.create(<Button>TEST</Button>);
    expect(cmp).toMatchSnapshot();
  });

  test("recieves className", () => {
    const cmp = renderer.create(<Button className="TEST_CLASS">TEST</Button>);
    expect(cmp).toMatchSnapshot();
  });

  test("disabled Button", () => {
    const cmp = renderer.create(<Button isDisabled>TEST</Button>);
    expect(cmp).toMatchSnapshot();
  });

  test("disabledClassName prop", () => {
    const cmp = renderer.create(
      <Button isDisabled disabledClassName="DISABLED_CLASS">
        TEST
      </Button>
    );
    expect(cmp).toMatchSnapshot();
  });

  test("triggers onClick", () => {
    const mockFn = jest.fn();
    const cmp = mount(<Button onClick={mockFn}>TEST</Button>);

    cmp.simulate("click");
    expect(mockFn).toHaveBeenCalled();
  });

  test("cannot click on disabled button", () => {
    const mockFn = jest.fn();
    const cmp = mount(
      <Button isDisabled={true} onClick={mockFn}>
        TEST
      </Button>
    );

    cmp.simulate("click");
    expect(mockFn).not.toHaveBeenCalled();
  });
});
