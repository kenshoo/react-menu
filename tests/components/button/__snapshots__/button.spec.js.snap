// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component disabled Button 1`] = `
<button
  className="button disabled"
  disabled={true}
>
  TEST
</button>
`;

exports[`Button component disabledClassName prop 1`] = `
<button
  className="button DISABLED_CLASS"
  disabled={true}
>
  TEST
</button>
`;

exports[`Button component recieves className 1`] = `
<button
  className="button TEST_CLASS"
  disabled={false}
>
  TEST
</button>
`;

exports[`Button component renders children 1`] = `
<button
  className="button"
  disabled={false}
>
  TEST
</button>
`;
